In [1]:
from transformers import AutoTokenizer, AutoModel

In [2]:
mod = AutoModel.from_pretrained('Qwen/Qwen2.5-0.5B')

config.json:   0%|          | 0.00/681 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/988M [00:00<?, ?B/s]

In [8]:
for n, p in mod.named_parameters():
    print(n, p.shape)

embed_tokens.weight torch.Size([151936, 896])
layers.0.self_attn.q_proj.weight torch.Size([896, 896])
layers.0.self_attn.q_proj.bias torch.Size([896])
layers.0.self_attn.k_proj.weight torch.Size([128, 896])
layers.0.self_attn.k_proj.bias torch.Size([128])
layers.0.self_attn.v_proj.weight torch.Size([128, 896])
layers.0.self_attn.v_proj.bias torch.Size([128])
layers.0.self_attn.o_proj.weight torch.Size([896, 896])
layers.0.mlp.gate_proj.weight torch.Size([4864, 896])
layers.0.mlp.up_proj.weight torch.Size([4864, 896])
layers.0.mlp.down_proj.weight torch.Size([896, 4864])
layers.0.input_layernorm.weight torch.Size([896])
layers.0.post_attention_layernorm.weight torch.Size([896])
layers.1.self_attn.q_proj.weight torch.Size([896, 896])
layers.1.self_attn.q_proj.bias torch.Size([896])
layers.1.self_attn.k_proj.weight torch.Size([128, 896])
layers.1.self_attn.k_proj.bias torch.Size([128])
layers.1.self_attn.v_proj.weight torch.Size([128, 896])
layers.1.self_attn.v_proj.bias torch.Size([128])

- 24 layers
- 151936 input tokens
- 869: size of each vectors. Hidden size of model

## Function calling: Ollama